common/includes/wporg-sso/wp-plugin.php

```diff
@@ -851,6 +872,53 @@ public function maybe_redirect_to_enable_2fa( $redirect, $orig_redirect, $user )
 			);
 		}
 
+		/**
+		 * Redirects the user to the 2FA Recovery codes nag if needed.
+		 */
+		public function maybe_redirect_to_recovery_codes( $redirect, $orig_redirect, $user ) {
+			if (
+				// No valid user.
+				is_wp_error( $user ) ||
+				// Or we're already going there.
+				str_contains( $redirect, '/recovery-codes' ) ||
+				// Or the user doesn't use 2FA
+				! Two_Factor_Core::is_user_using_two_factor( $user->ID )
+			) {
+				// Then we don't need to redirect to the enable 2FA page.
+				return $redirect;
+			}
+
+			// If the user logged in with a recovery code..
+			$session_token        = wp_get_session_token() ?: ( $this->last_auth_cookie['token'] ?? '' );
+			$session              = WP_Session_Tokens::get_instance( $user->ID )->get( $session_token );
+			$used_recovery_code   = str_contains( $session['two-factor-provider'] ?? '', 'Backup_Codes' );
+			$codes_available      = Two_Factor_Backup_Codes::codes_remaining_for_user( $user );
+
+			if (
+				// If they didn't use a recovery code,
+				! $used_recovery_code &&
+				(
+					// They have ample codes available..
+					$codes_available > 3 ||
```

Comment:
```suggestion
					$codes_available > 5 ||
```

This should be 5 to match the 2FA interface.

https://github.com/WordPress/wporg-two-factor/blob/4a154cc6a7e37a5acc28d467cff327cd80712e88/settings/src/components/backup-codes.js#L230

Reply:
I'm tempted to leave this at 3 despite the settings UI switching to a warning at 5, it makes sense that we'd nag on login later than we'd nag at them just casually browsing the settings UI.